src/coordinate_systems/PtEtaPhiM.jl

```diff
@@ -92,15 +92,9 @@ end
 #######################
 
 @inline function polar_angle(::PtEtaPhiM, mom)
-  xcomp = px(mom)
-  ycomp = py(mom)
-  zcomp = pz(mom)
-
-  return if iszero(xcomp) && iszero(ycomp) && iszero(zcomp)
-    zero(xcomp)
-  else
-    atan(transverse_momentum(mom), zcomp)
-  end
+  η = eta(mom)
```

Comment:
What is the reason for the change in the implementation? The new implementation performs a square root, which might cause inaccuracies.

Reply:
Sorry, maybe I'm misunderstanding, how is the new implementation performing a square root? The change for this implementation was because it was more succinct and did not require switching into Cartesian coordinates to find the polar angle, which feels a bit clunky. However, if this change makes things less accurate then there is no reason to keep things as they were.

Reply:
I understand your point about the cartesian coordinates, especially the `sincos` of `phi`.

The `spatial_magnitude` function performs the `sqrt`, which might lead to inaccuracies. However, I just realized, that it is just used in an `iszero` check, which means, just `iszero(pt(mom))` (which is the only way to make `r==0`) would also do the trick, right?

Reply:
Ah, I see what you are saying. This was done to copy the ROOT implementation of `polar_angle()`,  

` T Theta() const { using std::atan; return fRho > 0 ? 2 * atan(exp(-fEta)) : (fEta >= 0 ? 0 : pi()); }`

which you can find here https://root.cern/doc/master/GenVector_2CylindricalEta3D_8h_source.html#l00160

Having said this, I also see that what I wrote is slightly wrong insofar as it doesn't capture what to do when eta is greater or less than 0 like ROOT does. So maybe I need to change this if you think this definition is the one we want to use.

Reply:
Agreed 👍